### ISSUE TITLE
Empty list printing out as empty charlist in test assertion failure

### ISSUE THREAD

**Issue by sorliem** [elixir-lang/elixir#10453]
### Environment

* Elixir & Erlang/OTP versions (elixir --version): 
```
$ elixir --version
Erlang/OTP 21 [erts-10.3.5.6] [source] [64-bit] [smp:12:12] [ds:12:12:10] [async-threads:1] [hipe]

Elixir 1.10.4 (compiled with Erlang/OTP 21)
```
* Operating system: 
```
$ lsb_release -a
No LSB modules are available.
Distributor ID: LinuxMint
Description:    Linux Mint 19.3 Tricia
Release:        19.3
Codename:       tricia

$ uname -a
Linux lenovo-X1 5.4.0-48-generic #52~18.04.1-Ubuntu SMP Thu Sep 10 12:50:22 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```

### Current behavior

Given this failing example test, it prints the value of foo - an empty list `[]`, as an empty charlist `''`:

```elixir
defmodule TestProjectTest do
  use ExUnit.Case

  test "show odd printing behaviour" do
    a = %{foo: [], bar: []}
    b = %{foo: [], bar: [1]}

    assert a == b
  end
end
```

```
$ mix test


  1) test show odd printing behaviour (TestProjectTest)
     test/test_project_test.exs:4
     Assertion with == failed
     code:  assert a == b
     left:  %{bar: [], foo: ''}
     right: %{bar: [1], foo: ''}
     stacktrace:
       test/test_project_test.exs:8: (test)



Finished in 0.01 seconds
1 test, 1 failure

Randomized with seed 786571
```

### Expected behavior

I would expect the left/right output to have the value of `foo` be printed out as `[]`, not `''`.

**Comment by ityonemo** [comment]
I've noticed this too, that I am never really sure when it's going to pick `[]` vs `''`

**Comment by smaximov** [comment]
> I would expect the left/right output to have the value of foo be printed out as [], not ''

I don't think it's possible though because there's no way to tell if a value is an empty charlist or an empty list &mdash; they are in fact different literals representing the same value.

**Comment by eksperimental** [comment]
> > I would expect the left/right output to have the value of foo be printed out as [], not ''
> 
> I don't think it's possible though because there's no way to tell if a value is an empty charlist or an empty list — they are in fact different literals representing the same value.

The thing is that it is not consistent. Empty lists are represented as `''` and `[]`. It shoud behave the way `Kernel.inspect/2` does

**Comment by smaximov** [comment]
Ah, thanks, I got your point.

**Comment by eksperimental** [comment]
I can give a try trying to fix this if nobody is working on it

**Comment by eksperimental** [comment]
This has been fixed in v1.11,
tested in v1.11 and in master
I can see it is happening in v1.10.4

I guess the issue can be closed.

```
$ elixir --version
Erlang/OTP 23 [erts-11.0.4] [source] [64-bit] [smp:8:8] [ds:8:8:10] [async-threads:1] [hipe]

Elixir 1.11.0 (compiled with Erlang/OTP 21)

  1) test show odd printing behaviour (ExUnitEmptyListsTest)
     test/ex_unit_empty_lists_test.exs:9
     Assertion with == failed
     code:  assert a == b
     left:  %{bar: [], foo: []}
     right: %{bar: [1], foo: []}
     stacktrace:
       test/ex_unit_empty_lists_test.exs:13: (test)

Finished in 0.03 seconds
1 doctest, 2 tests, 1 failure

Randomized with seed 720737
```

**Comment by sorliem** [comment]
> This has been fixed in v1.11

Cool with me! Closing.
